adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -390,7 +390,7 @@ protected void uploadMessage(final AmqpContext context) {
             if (tenantObject.isAdapterEnabled(getTypeName())) {
 
                 final MessageSender sender = senderFuture.result();
-                final ResourceIdentifier resource = ResourceIdentifier.from(endpointName, context.getTenantId(), deviceId);
+                final ResourceIdentifier resource = ResourceIdentifier.from(endpointName, context.getTenantId(), context.getDeviceId());
```

Comment:
why not simply use `context.getResourceIdentifier()` instead of instantiating a new one?